tests/v2_validation/cattlevalidationtest/core/test_ldap_auth.py

```diff
@@ -1590,3 +1592,126 @@ def test_secret_setting(admin_client):
+        r = requests.post(auth_url, data=json.dumps(config),
+                          auth=(access_key, secret_key))
+        assert r.ok
+    else:
+        ADMIN_AD_CLIENT.create_ldapconfig(config)
+
+    cookies = dict(token=token3['jwt'])
+    bad_auth = requests.get(cattle_url() + "schemas", cookies=cookies)
+    assert bad_auth.status_code == 403
+
+    cookies = dict(token=token2['jwt'])
+    good_auth = requests.get(cattle_url() + "schemas", cookies=cookies)
+    assert good_auth.status_code == 200
+
+    config = load_config()
+    config['enabled'] = None
+    if ldap_port == 'True':
+        auth_url = cattle_url()[:-7] + 'v1-auth/config'
+        r = requests.post(auth_url, data=json.dumps(config),
+                          auth=(access_key, secret_key))
+        assert r.ok
+    else:
+        client = create_ad_client(username=ldap_main_user,
+                                  password=ldap_main_pass)
+        client.create_ldapconfig(config)
+
+
+# 3
+@if_test_ad
+def test_ad_restrict_to_specific_group(admin_client):
+    ldap_main_user = os.environ.get('AD_MAIN_USER')
+    ldap_main_pass = os.environ.get('AD_MAIN_PASS')
+    ldap_user2 = os.environ.get('AD_USER2')
+    ldap_pass2 = os.environ.get('AD_PASS2')
+    ldap_user3 = os.environ.get('AD_USER3')
+    ldap_pass3 = os.environ.get('AD_PASS3')
+    ldap_port = os.environ.get('LDAP_PORT')
+    group = os.environ.get('AD_GROUP')
+
+    main_client = create_ad_client(username=ldap_main_user,
+                                   password=ldap_main_pass)
+    user = ADMIN_TOKEN['userIdentity']
+    allowed_identities = []
+    allowed_identities.append(user)
+    token2 = get_authed_token(username=ldap_user2,
+                              password=ldap_pass2)
+    token3 = get_authed_token(username=ldap_user3,
+                              password=ldap_pass3)
+
+    group_identity = main_client.list_identity(name=group)[0]
+    group_identity_dict = ast.literal_eval(str(group_identity))
+    allowed_identities.append(group_identity_dict)
+    # Enable new configuration
+    config = load_config(access_mode='required')
+    config['enabled'] = True
+    config['allowedIdentities'] = allowed_identities
+
+    if ldap_port == 'True':
+        access_key = ADMIN_AD_CLIENT._access_key
+        secret_key = ADMIN_AD_CLIENT._secret_key
+        auth_url = cattle_url()[:-7] + 'v1-auth/config'
+        r = requests.post(auth_url, data=json.dumps(config),
+                          auth=(access_key, secret_key))
+        assert r.ok
+    else:
+        ADMIN_AD_CLIENT.create_ldapconfig(config)
+
+    cookies = dict(token=token3['jwt'])
+    bad_auth = requests.get(cattle_url() + "schemas", cookies=cookies)
+    assert bad_auth.status_code == 403
+
+    cookies = dict(token=token2['jwt'])
+    good_auth = requests.get(cattle_url() + "schemas", cookies=cookies)
+    assert good_auth.status_code == 200
+
+    config = load_config()
+    config['enabled'] = None
+    if ldap_port == 'True':
```

Comment:
Same as above test case, no need to disable auth, we should only set it to unrestricted in a finalizer